qiskit/backends/ibmq/ibmqbackend.py

```diff
@@ -155,22 +157,19 @@ def jobs(self, limit=50, skip=0, status=None):
         """
         backend_name = self.configuration['name']
         job_list = []
-        base_index = skip
-        job_info_list = []
         if isinstance(status, str):
             status = JobStatus[status]
-        while len(job_list) < limit or len(job_info_list) < limit:
-            job_info_list = self._api.get_jobs(limit=limit, skip=base_index,
-                                               backend=backend_name)
-            for job_info in job_info_list:
-                is_device = not bool(self._configuration.get('simulator'))
-                job = IBMQJob.from_api(job_info, self._api, is_device)
-                if len(job_list) < limit:
-                    if status is None:
-                        job_list.append(job)
-                    elif job.status.get('status') == status:
-                        job_list.append(job)
-            base_index += limit
+
+        job_info_list = self._api.get_jobs(limit=limit, skip=skip,
+                                           backend=backend_name)
+        for job_info in job_info_list:
+            is_device = not bool(self._configuration.get('simulator'))
+            job = IBMQJob.from_api(job_info, self._api, is_device)
+            if len(job_list) < limit:
+                if status is None:
+                    job_list.append(job)
+                elif job.status.get('status') == status:
+                    job_list.append(job)
```

Comment:
The reason for the while loop and incrementing base_index (which I agree probably calls the api too much) is so that the call returns the number of jobs they ask for if that number of jobs satisfying the status condition exist on the server. For instance if someone wants there last 20 jobs with errors and they call `backend.jobs(limit=20, status=JobStatus.ERROR)` then they might not get any of their errored jobs. I think they should get 20 errored jobs if the server has them.

Reply:
Ok, got it. This is definitely something that should be filtered at the server level. We can't afford getting all the list of jobs to find out that the status we are looking for matches the last job... after retrieving hundreds of them. I'll talk with the server guys :)

Reply:
That would be much better. They may have database filters which are much faster than what is being done here -- they recently did the backend filtering. I think status and date would also be interesting. Maybe more?